liquibase-standard/src/main/java/liquibase/command/core/GenerateChangelogCommandStep.java

```diff
@@ -158,9 +158,12 @@ public void run(CommandResultsBuilder resultsBuilder) throws Exception {
                     changeLogWriter.print(outputStream);
                 }
             }
-            if (StringUtil.trimToNull(changeLogFile) != null) {
+            if (!changeLogWriter.generateChangeSets().isEmpty()) {
                 Scope.getCurrentScope().getUI().sendMessage("Generated changelog written to " + changeLogFile);
             }
+            else {
+                Scope.getCurrentScope().getUI().sendMessage(String.format("Changelog not generated. There are not changesets to write to %s changelog", changeLogFile));
```

Comment:
There is a small typo here. Message should be:
```
"Changelog not generated. There are no changesets to write to %s changelog"
```

Reply:
Well spotted, corrected now, will be included in next commit.

Reply:
Implemented and pushed.